subprojects/core/src/integTest/groovy/org/gradle/cache/internal/DefaultGeneratedGradleJarCacheIntegrationTest.groovy

```diff
@@ -52,7 +53,7 @@ class DefaultGeneratedGradleJarCacheIntegrationTest extends Specification {
     def services = (DefaultServiceRegistry) ServiceRegistryBuilder.builder()
             .parent(NativeServicesTestFixture.getInstance())
             .provider(LoggingServiceRegistry.NO_OP)
-            .provider(new GlobalScopeServices(false))
+            .provider(new GlobalScopeServices(false, new DisabledAgentStatus()))
```

Comment:
🎨 I'd avoid spreading the knowledge about how to configure the agent status with something like [this](https://github.com/gradle/gradle/pull/23821/commits/cd571431de62f79efa48e35cd998c2db92c472e0).

Reply:
I tweaked the logic a bit, following your idea. However, I'm not sure if having the implicit "agent disabled" behavior for `GlobalScopeServices` is a good thing. I'd rather encourage the caller of the constructor to think about the agent status for the process being initialized. Unfortunately, we have the `GlobalScopeServices` in too many use cases.